src/sdk/namespace_fs.js

```diff
@@ -665,7 +667,8 @@ class NamespaceFS {
                 /**
                  * @typedef {{
                  *  key: string,
-                 *  common_prefix: boolean
+                 *  common_prefix: boolean,
+                 *  pos: string,
```

Comment:
I saw on the PR description you mentioned that you tried to avoid this next_pos and put this value on next_marker as the original solution suggested but you changed it because ceph tests were failing.
But existing ceph tests should not fail because ALLOW_NSFS_UNSORTED_OBJECTS_LIST should be false.
Have you tried to revert it? we should try to solve the issue so we can use marker/next_marker

Reply:
@romayalon checking this